pkgs/misc/vim-plugins/overrides.nix

```
@@ -810,6 +810,16 @@ self: super: {
     dependencies = with self; [ vim-maktaba ];
   });
 
+  # Due to case-sensitivety issues, the hash differs on Darwin systems, see:
+  # https://github.com/NixOS/nixpkgs/issues/157609
+  vim-colorschemes = super.vim-colorschemes.overrideAttrs (old: {
+    src = old.src.overrideAttrs (_: {
+      sha256 = if stdenv.isDarwin
+        then "0kpsf6j20fgblc8vhqn7ymr52v2d1h52vc7rbxmxfwdm80nvv3g5"
+        else "1cg8q7w0vgl73aw1b9zz0zh5vw5d2pm8pm54fhfzva4azg56f416";
```

can we just remove the file in question?

```suggestion
      postUnpack = ''
        rm colors/darkBlue.vim
      '';
```

---

the hash will need to be regenerated

---

another option could be to specify the pr which fixes the issue

```
flazz/vim-colorschemes@refs/pull/181/head
```

---

Regarding removing the file, I'm not sure. I tried making the change, and setting the hash to `""` in `generated.nix` so that `nix` would tell me which hash it was expecting, and it continued to expect the two different hashes on my Darwin and Linux systems respectively. Maybe I'm confused ¯\_(ツ)_/¯.

Regardless between doing that and switching to the PR ref, I think I'd prefer using the PR ref.

However, I think it might be the case the current solution in the PR is the best way to go. Here's my thinking (which might be missing something):
* If using `postUnpack` works, but then the PR is later merged on the upstream repo, the overlay will hang around like a barnacle until someone notices it and realizes it can be removed. I have some esthetic that doesn't like this cruft accumulating silently.
* If we switch to using the PR ref, but then the owner of the repo becomes active again and merges the PR and makes other updates, someone will have to notice this and update the `vim-plugin-names` file to point to the `master` branch again. The same esthetic from the last bullet doesn't like that the plugin could become silently out of date if this happens.
* If we use the current proposed solution, but then the owner of the plugin becomes active again and adds any new commits, `nix` will complain of a hash mismatch, and the only solution will be to delete or update the overlay (depending on whether the original issue was fixed, likely by the upstream PR being merged).

Thoughts?

---

@jonringer, no particular rush, but just wanted to poke you to make sure this thread didn't get dropped.

---

Just checking in again.

@SuperSandro2000, maybe you could help?

---

I would really like to avoid platform-dependent hashes, as it's prone to breakages.

---

> * If we use the current proposed solution, but then the owner of the plugin becomes active again and adds any new commits, `nix` will complain of a hash mismatch, and the only solution will be to delete or update the overlay (depending on whether the original issue was fixed, likely by the upstream PR being merged).

It will only if the current hash is not available which it will be for a while.

---

> * If using `postUnpack` works, but then the PR is later merged on the upstream repo, the overlay will hang around like a barnacle until someone notices it and realizes it can be removed. I have some esthetic that doesn't like this cruft accumulating silently.

If we do rm on the file without -f then the build will fail as soon as the file is no longer available.

---

@SuperSandro2000, @jonringer, alrighty, I finally got back around to this.

As far as I can tell, when overriding `src` the attributes we have to work with are the result of calling `fetchurl` (which is called from `fetchzip` which itself is called by `fetchFromGitHub`). So `fetchurl` doesn't use the generic builder, and its builder doesn't have an `unpack` phase let alone a `postUnpack` phase, the only thing we have to work with is its `postFetch` phase.

I got things working by appending `rm $out/colors/darkBlue.vim` to `postFetch` on non-`darwin` systems. We can't also run that command on `darwin` systems, because it removes `darkblue.vim` (without the capital **B**) due to case-insensitivity.

I've tested on both my macOS machine as well as my NixOS machine, and the hashes now matched on both.